sdl_viewer/src/main.rs

```diff
@@ -40,8 +43,64 @@ use std::process;
 use std::ptr;
 use std::str;
 
-const FRAGMENT_SHADER: &'static str = include_str!("../shaders/points.fs");
-const VERTEX_SHADER: &'static str = include_str!("../shaders/points.vs");
+const FRAGMENT_SHADER_POINTS: &'static str = include_str!("../shaders/points.fs");
+const VERTEX_SHADER_POINTS: &'static str = include_str!("../shaders/points.vs");
+
+fn draw_octree_debug_view(_outlined_box_drawer: &OutlinedBoxDrawer, _camera: &Camera, _camera_octree: &Camera, _visible_nodes: &Vec<octree::VisibleNode>, _node_views: &mut NodeViewContainer)
+{
+    unsafe {
+        let x = _camera_octree.width;
+        let y = 0;
+        gl::Viewport(x, y, _camera_octree.width, _camera_octree.height);
+        gl::Scissor(x, y, _camera_octree.width, _camera_octree.height);
+        gl::Enable(gl::SCISSOR_TEST);
+
+        gl::ClearColor(0.3, 0.3, 0.4, 1.0);
+        gl::Clear(gl::COLOR_BUFFER_BIT | gl::DEPTH_BUFFER_BIT);
+    }
+
+    let mx_camera_octree: Matrix4<f32> = _camera_octree.get_world_to_gl();
+    for visible_node in _visible_nodes {
+        if let Some(_view) = _node_views.get(&visible_node.id) {
+            let color = vec![1.,1.,0.,1.];
+            draw_outlined_box(&_outlined_box_drawer, &mx_camera_octree, _view, &color);
+        }
+    }
+    
+    // frustum
+    let color = vec![1.,0.,1.,1.,1.];
+    _outlined_box_drawer.update_color(&color);
+    let mx_inv_camera:  Matrix4<f32> = _camera.get_world_to_gl().inverse_transform().unwrap().into();
+    let mx = mx_camera_octree * mx_inv_camera;
+    _outlined_box_drawer.update_transform(&mx);
+    _outlined_box_drawer.draw();
```

Comment:
why does not the draw() function take color and transform as parameters? You always call both `update` functions, then call draw() anyways?